src/geometry/rotation.rs

```diff
@@ -154,12 +154,7 @@ impl<T: Scalar, const D: usize> Rotation<T, D> {
     /// assert_eq!(*rot.matrix(), mat);
     /// ```
     #[inline]
-    pub fn from_matrix_unchecked(matrix: SMatrix<T, D, D>) -> Self {
-        assert!(
-            matrix.is_square(),
-            "Unable to create a rotation from a non-square matrix."
-        );
```

Comment:
Note that squareness is still statically enforced, since we take `SMatrix<_, D, D>`, which doesn't allow for dynamic sizing.